### ISSUE TITLE
Window position reset to center when viewing next image

### ISSUE THREAD

**Issue by webstory** [linuxmint/xviewer#106]
```
 * Xviewer version 2.6.2
 * Distribution - Linux Mint 20 Cinnamon 64-bit
```

**Issue**
Window position reset to center when go to the prev/next image.

**Steps to reproduce**
 - Load one image
 - Move window (Drag-drop title bar)
 - Press Left/Right Arrow key

**Expected behaviour**
 - Window position and size are fixed.

**Other information**
 - This bug does not appeared an older version of XViewer (probably 2.4.x)

**Comment by programmer-ceds** [comment]
Apologies - this was a side-effect of the change that I made to remember the maximized state from one xviewer activation to the next. There are two functions in xviewer-window.c that maximize or unmaximize the image each time they are called and this was causing the problem. I have tested a change that only alters the maximization state when necessary and this restores the original behaviour (the moved, and optionally resized, image window stays put when you move to the next or previous image).

I will include the corrected code in a new pull request once the problem with PR 101 is resolved.

**Comment by programmer-ceds** [comment]
Following the merging of PR #112 this issue can be closed.

**Comment by jsarasin** [comment]
Any idea how long until this fix it out in the updates for LM?

**Comment by programmer-ceds** [comment]
You can use the latest unstable release that includes the latest changes.

Go to https://github.com/linuxmint/xviewer

Towards the right-hand side there is a "Releases" heading with a "Latest" button - click the "Latest" button.

Download packages.tar.gz to a temporary folder.

Unzip packages.tar.gz to give you packages.tar - unzip this to extract a folder titled "packages" - go into this folder with nemo

Open a terminal window in the "packages" folder

Type:

sudo dpkg -i xviewer_2.6.2_amd64.deb

You will probably get the following warning but I haven't found that this caused any problems.

dpkg: warning: downgrading xviewer from 2.6.2+ulyana to 2.6.2

**Comment by programmer-ceds** [comment]
@jsarasin 
I suspect that there is a way to get the latest unstable version using the "Releases/Latest" link on the home page (https://github.com/linuxmint/xviewer) but trying this route I get the problem described in issue 117. Below is the method that works for me to get the latest version or to install a version that I have edited (and not get the broken package error from the Update Manager)

Using Linux Mint 20 Cinnamon:

(**** Ignore all quote marks in the notes below ****)

If you haven't already installed the Mint development tools you will need to open a terminal window and type the following:

"apt update"
"apt install mint-dev-tools"
"mint-dev-setup"

The mint-dev-setup step will ask a number of questions including requesting you GIT username, email etc. All of these questions can be answered by simply pressing the Enter key (which will also select any default options)

(The above steps have to be done only once)

Use a web browser to go to the xviewer home page https://github.com/linuxmint/xviewer

Left-click on the green "Code" button
Left-click on the clipboard icon to the right of the https address

In nemo browse to a temporary folder (create one if necessary).
Right-click in the temporary folder and select "Open in Terminal"
Type "git clone " then right-click and select "Paste".
Press Enter - GIT should clone the xviewer files into a sub-folder called xviewer in the temporary folder.
Type "mint-build"

Once this step has completed (it may appear to hang at times) ignore the package files that it has created in the temporary folder (see issue 117)

Type:

"cd xviewer"
"sudo make install"

Once the make install step finishes you should have the latest (unstable) version of xviewer installed. You can close the terminal window and, if you wish, delete the temporary folder.

Should you wish to make any edits to the code the new version of xviewer can be generated quicker by opening a terminal window in the xviewer folder that was created by the GIT clone operation and typing:

"make"
"sudo make install"

Note that this will not work with freshly cloned files when the mint-build path must be followed.

Hope this helps.

**Comment by jsarasin** [comment]
right on! thanks for the detailed reply!
I've been kinda thinking about contributing to Mint.  I'd like to help out with the display positioning (multiple monitors) This will get me going in the right direction.
